### ISSUE TITLE
External Storage Support Local: Sharing folder with other Tools e.g. resilio

### ISSUE THREAD

**Issue by mkriegl** [nextcloud/server#16844]
Hi guys,

I am using nextcloud in parallel with resilio-sync on one raspberry. My plan was to share one folder between nextcloud and resilio-sync via external storage support. This is useful for me, when moving files via resilio to my mobile devices or NAS for archiving. Moving the file out of the folder on the other device will make the file also disappear in the local folder. 
Therefore I added a folder on the raspberry with chmod 770, chown www-data:www-data, included the folder as external local storage support in Nextcloud's admin console and added the user for the resilio-sync to the group www-data.
My problem is, that the files being added to that folder via Nextcloud always get the chmod 750 or 740 credentials. Therefore, I am able to make resilio sync the files in that folder to my mobile device or NAS, but when moving the files out of the folder (NAS or mobile) resilio is not able to remove the files on the raspberry, because of missing write access rights. 

1. Is it possible to change the chmod access rights for files being written in the external local storage support folder
2. Can probably a function be added as a setting to the external storage support folder for selection of write access to www-data group

I would be fine when No.1 only would be possible, but I guess including No.2 would be more comfortable to have folders with different settings.

Thanks in advance .. Max

**Comment by kesselb** [comment]
Hmm. 750 or 740 is still group readable. Have you tried to add the user running resilio-sync to the www-data group?

**Comment by mkriegl** [comment]
Probably I wrote it to complicated :-)
The user resilio has been added to the group www-data
The problem is, that resilio needs write access to those files. Nextcloud saves files only with read access to the group, so there is no chance to handle those files via resilio and I get an error/warning within resilio not having write access. :-)

**Comment by kesselb** [comment]
https://github.com/nextcloud/server/blob/1cc8a2f5d251b5f2ab594f9c96eaff77ce6697f3/lib/private/Files/Storage/Local.php

There is no config option to change the chmod. Actually we don't call chmod for local storage. So the files are created with the default permissions. Could you try to adjust the default umask?

**Comment by mkriegl** [comment]
Sorry for your inconvenience, but after some testing I got the issue. It is not within the group rights, it is more with other rights and resilio sync has therefore an issue with it. Those must be readable as a minimum. I guess the error occurred, as I set a folder to 660. After moving files into Nextcloud and those being synced as usual with the other server I recognised the access right 664 on both servers. When moving a file with 660 to the sync folder I get errors in both resilio sync applications and the file is not transferred. 
So, in the end the only thing to do is to add the resilio-sync user to the www-data group. And take care that the folders and files do have minimum read rights for others. 
Still thanks for your help
